MigrationTutorial/Models/V2/Supplier.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using MongoDB.Bson;
+using Realms;
+
+namespace MigrationTutorial.Models.V2
+{
+    public class Supplier : RealmObject
+    {
+        [PrimaryKey]
+        public ObjectId Id { get; private set; } = ObjectId.GenerateNewId();
+
+        public string Name { get; set; }
+
+        //[Required]
+        private ISet<string> _SuppliedTypes { get; }
+
+        [Ignored]
+        public ISet<ConsumableType> SuppliedTypes { get; } = new HashSet<ConsumableType>();
+
+        public Supplier() { }
+
+        public void AddConsumableType(ConsumableType[] consumables)
```

Comment:
Unfortunately I don't think it's possible in other way. A thing to notice is that this is inherently dangerous anyway, as there is no way to prevent the user of this class to do "illegal" changes, such as removing an element from SuppliedTypes without updating _SuppliedTypes too.

Reply:
Mmmh, maybe the only other way would be to create a class that derives from HashSet and that overrides Add and Remove and does some magic, but it seems messy.

Reply:
What one could do is to re-implement the `ISet` interface to make your own collection. Such collection could be string backed but `Contains`, `Add` and `Remove` would take an enum which behind the scene is converted to a string before being used as a key. But beside being more "ergonomic" I don't think there's a lot more to it. If I was writing production code, I'd likely do that, of course. For the sake of speed adding those 4 methods were faster and still show the same concept.

> Something similar to overwriting a setter and getter

You need to remember that `set` for a collection simply changes what collection the var points to. At least in the standard .NET. Changing this behaviour is generally not discouraged. What did you have in mind, exactly?

Reply:
Oh yeah, I know. I probably did not explain it good enough. I just meant overwriting in general. Not the `setter` and `getter` but the `adder` and `remover` if that makes any sense. :D

Reply:
I guess having an interface like described above might be nice but then again the fields are private anyway so in terms of safety we should be ok here. It was more about 'how to make sure we don't mix anything up because we always have to update both fields'?

Reply:
@papafe, again page not refreshed. I wonder if the browser should auto refresh comments or not on GH.
Anyway, you're right @papafe, a person could easily mess up the loose connection between the 2 fields. I'd say that I could explain in the blog-post why this was done instead of implement my own collection and the dangers my approach could introduce. But if you guys think that just this is bad, then I'll make my own collection. I'd just have liked to avoid spending too much time on accessory parts and focus more on the migration itself.